numpy/core/src/multiarray/temp_elide.c

```diff
@@ -363,6 +363,8 @@ can_elide_temp_unary(PyArrayObject * m1)
     if (Py_REFCNT(m1) != 1 || !PyArray_CheckExact(m1) ||
             !PyArray_ISNUMBER(m1) ||
             !(PyArray_FLAGS(m1) & NPY_ARRAY_OWNDATA) ||
+            !PyArray_ISWRITEABLE(m1) ||
+            PyArray_CHKFLAGS(m1, NPY_ARRAY_UPDATEIFCOPY) ||
```

Comment:
This line seems to be an unrelated bugfix (with no test?)

Reply:
Both fixes went in together for binary functions. I suspect the updateifcopy check is no longer needed but would like to keep it in until those types of arrays are completely gone. However, they are gone enough that I no longer know how to test it as `a.flat.__array__()` now returns a straight copy.

Reply:
Also can you change the `PyArray_FLAGS` above to also use PyArray_CHKFLAGS for consistency?